Assets/MixedRealityToolkit.SDK/Features/Utilities/StabilizationPlaneModifier.cs

```diff
@@ -0,0 +1,390 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Input;
+using System;
+using UnityEngine;
+using UnityEngine.XR.WSA;
+
+namespace Microsoft.MixedReality.Toolkit.Utilities
+{
+    [Serializable]
+    public struct StabilizationPlaneOverride
+    {
+        public Vector3 Center;
+        public Vector3 Normal;
+    }
+
+    /// <summary>
+    /// StabilizationPlaneModifier handles the setting of the stabilization plane in several ways.
+    /// Using StabilizationPlaneModifier with override DepthLSR which is automatically enabled via depth buffer sharing
+    /// StabilizationPlaneModifier is recommended for HoloLens 1, can be used for HoloLens 2, and does a no op for WMR
+    /// </summary>
+    public class StabilizationPlaneModifier : MonoBehaviour
+    {
+        [System.Serializable]
+        public enum StabilizationPlaneMode
+        {
+            Fixed,
+            TargetOverride,
+            PlaneOverride,
+            GazeHit,
+            Off
+        }
+
+        [Tooltip("Choose mode for stabilization plane.\n1) Fixed- Submits plane at a fixed distance based on DefaultPlaneDistance field along the users gaze.\n2) Gaze Hit- Submits plane along the users gaze at the position of gaze hit.\n3) Target Override- Submits the plane at a fixed position along the users gaze based on the TargetOverride property.\n4) Plane Override- Submits the plane based on the OverridePlane property.\n5) Off- Does not call SetFocusPoint")]
+        public StabilizationPlaneMode Mode;
+
+        [Tooltip("When lerping, use unscaled time. This is useful for apps that have a pause mechanism or otherwise adjust the game timescale.")]
+        public bool UseUnscaledTime = true;
+
+        [Tooltip("Lerp speed when moving focus point closer.")]
+        public float LerpStabilizationPlanePowerCloser = 4.0f;
+
+        [Tooltip("Lerp speed when moving focus point farther away.")]
+        public float LerpStabilizationPlanePowerFarther = 7.0f;
```

Comment:
Nit: Here's a weird thing that I don't necessarily agree with but.... in order to be consistent with our other code, I think that we've moved away from having public fields, in favor of having private fields (like targetOverride) and public getters and setters.

I don't necessarily agree with this overall, but I would like for us to be consistent so absent us taking a large uturn on that policy I think it would be good to remain consistent with it.

So specifically what I'm asking for here is to make these private, add a public getter/setter, and then add the SerializeField property on the newly privated fields (so that they can keep getting serialized)

Sorry about this